### PR TITLE
Function tests cleanup (part1)

### PR DESCRIPTION
**Pull request by kichalla**
This PR addresses part of https://github.com/aspnet/Mvc/issues/3624 ...will send out a separate PR out for the rest of the changes.

@rynowak @dougbu 

Filed tracking issue: https://github.com/aspnet/Mvc/issues/3670

### CODE REVIEW

File: Mvc.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
-VisualStudioVersion = 14.0.24711.0
+VisualStudioVersion = 14.0.24627.0
```
Comment (pranavkm) [inline Mvc.sln]: Tsck tsck. You need to update VS

**Comment by dougbu** [comment]
:shipit: from me but get @rynowak's squirrel too

File: test/Microsoft.AspNet.Mvc.FunctionalTests/RequestServicesTest.cs [hunk 2]
```diff
@@ -21,11 +21,10 @@ public RequestServicesTest(MvcTestFixture<RequestServicesWebSite.Startup> fixtur
...
         [Theory]
-        [InlineData("http://localhost/RequestScoped/FromController")]
-        [InlineData("http://localhost/Other/FromFilter")]
-        [InlineData("http://localhost/Other/FromView")]
-        [InlineData("http://localhost/Other/FromViewComponent")]
-        [InlineData("http://localhost/Other/FromActionArgument")]
+        [InlineData("http://localhost/RequestScopedService/FromFilter")]
+        [InlineData("http://localhost/RequestScopedService/FromView")]
+        [InlineData("http://localhost/RequestScopedService/FromViewComponent")]
+        [InlineData("http://localhost/RequestScopedService/FromActionArgument")]
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/RequestServicesTest.cs]: We seem to have lost a test case here... what gives?

File: test/Microsoft.AspNet.Mvc.FunctionalTests/TempDataTest.cs [hunk 3]
```diff
@@ -186,10 +186,10 @@ public TempDataTest(MvcTestFixture<TempDataWebSite.Startup> fixture)
             // Act & Assert
             var exception = await Assert.ThrowsAsync<InvalidOperationException>(async () =>
             {
-                await Client.PostAsync("/Home/SetTempDataInvalidType", content);
+                await Client.PostAsync("/TempData/SetTempDataInvalidType", content);
             });
             Assert.Equal("The '" + typeof(SessionStateTempDataProvider).FullName + "' cannot serialize an object of type '" +
-                typeof(TempDataWebSite.Controllers.HomeController.NonSerializableType).FullName + "' to session state.", exception.Message);
+                typeof(BasicWebSite.Controllers.TempDataController.NonSerializableType).FullName + "' to session state.", exception.Message);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/TempDataTest.cs]: We should delete this test. We don't need a functional test to verify that we throw an exception here.

**Comment by rynowak** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/RequestServicesTest.cs (hunk 2 above)
Comment (kichalla) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/RequestServicesTest.cs]: Good catch...and it was intentional as the scenario of `FromController` is not required as its covered by `FromActionArgument` now...we recently removed support for `FromServices` on a property which `FromController` was testing... https://github.com/aspnet/Mvc/pull/3671/files#diff-b2fc13d7d7ddd5d3ee5929cf0af23cc5L12

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/RequestServicesTest.cs]: Ok sounds good.

**Comment by kichalla** [comment]
Updated @rynowak

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
dd774366f6fb18af9382ca84c276e6eb73c66b3a

**Pull request closed without merging**